src/libraries/System.Net.Http/src/System/Net/Http/FormUrlEncodedContent.cs

```diff
@@ -28,30 +28,61 @@ private static byte[] GetContentByteArray(IEnumerable<KeyValuePair<string?, stri
             ArgumentNullException.ThrowIfNull(nameValueCollection);
 
             // Encode and concatenate data
-            StringBuilder builder = new StringBuilder();
+            var builder = new ValueStringBuilder(stackalloc char[256]);
+
             foreach (KeyValuePair<string?, string?> pair in nameValueCollection)
             {
                 if (builder.Length > 0)
                 {
                     builder.Append('&');
                 }
 
-                builder.Append(Encode(pair.Key));
+                Encode(ref builder, pair.Key);
                 builder.Append('=');
-                builder.Append(Encode(pair.Value));
+                Encode(ref builder, pair.Value);
             }
 
-            return HttpRuleParser.DefaultHttpEncoding.GetBytes(builder.ToString());
+            // We know the encoded length because the input is all ASCII.
```

Comment:
How do we know the input is ASCII?

Reply:
The input is a bunch of concatenated `Uri.EscapeDataString` results, which are always ASCII, and the encoding is Latin1.
I clarified it in the comment now.